packages/calypso-e2e/src/browser-helper.ts

```diff
@@ -21,31 +27,59 @@ export function getViewportName(): viewportName {
  * If the environment variable BROWSERLOCALE is set, this will override all configuration
  * values. Otherwise, the default path contained in the configuration file is returned.
  *
- * @returns {localeCode} Target locale code.
+ * @returns {LocaleCode} Target locale code.
  */
-export function getLocale(): localeCode {
-	return ( process.env.LOCALE || config.get( 'locale' )! ).toLowerCase() as localeCode;
+export function getLocale(): LocaleCode {
+	return ( process.env.LOCALE || config.get( 'locale' )! ).toLowerCase() as LocaleCode;
 }
 
 /**
- * Returns a set of screen dimensions in numbers.
- *
- * This function takes the output of `getViewportName` and returns an
- * object key/value mapping of the screen diemensions represented by
- * the output.
+ * Returns launch configuration of a device mapping to the target device.
  *
- * @param {viewportName} [target] Target display size to use, overriding defaults.
- * @returns {viewportSize} Object with key/value mapping of screen dimensions.
- * @throws {Error} If target screen size was not set.
+ * @param {TargetDevice} name Name of the target device.
+ * @returns {BrowserContextOptions} Launch configuration of the device.
  */
-export function getViewportSize( target?: viewportName ): viewportSize {
-	if ( ! target ) {
-		target = getViewportName();
+export function getDevice( name: TargetDevice ): BrowserContextOptions {
+	if ( name === 'mobile' ) {
+		return devices[ 'Pixel 4a (5G)' ];
 	}
-
-	const sizes: { [ key: string ]: viewportSize } = config.get( 'viewportSize' );
-	if ( ! Object.keys( sizes ).includes( target ) ) {
-		throw new Error( `Unsupported viewport: ${ target }` );
+	if ( name === 'desktop' ) {
+		return devices[ 'Desktop Chrome HiDPI' ];
 	}
-	return sizes[ target ];
+	throw new Error( 'Unknown target device.' );
```

Comment:
This is where the target device themselves are obtained from Playwright's list of devices, which is [here](https://github.com/microsoft/playwright/blob/97717f25d2e1f317c5c656879f0aa90bd3873478/src/server/deviceDescriptorsSource.json).